### ISSUE TITLE
Reproducible black screen (hang ?) after ~15' on old K8 platform

### ISSUE THREAD

**Issue by debrouxl** [memtest86plus/memtest86plus#320]
I recently bought a 2S Opteron Italy platform: RioWorks HDAMA, 2 x Opteron 270 2C/2T @ 2 GHz, 8 x 1 GB PC-2100R RAM in 2 x 2C, 4 GB NUMA nodes.
memtest86+ starts, and no matter what memtest86+ does - even wait in the initial configuration menu - the screen suddenly becomes black a bit more than 15' after starting the program. Pressing ESC does not reboot the computer, I have to use the power button.
This happens with both 6.10 and the current HEAD of the numa branch, b7da80f4ed501715f8e2aba8bd398c68c8a669e2 ; whether HPET is enabled and exposed by the ACPI BIOS does not matter, either.

Under Linux, the EDAC spews an unending stream of ECC corrected errors every 5'10" - 5'11". The 600+ occurrences which have appeared so far in logwatch daily messages belong to the ranges of two of the RAM sticks, both on CPU 2, and encompass many different addresses, though some occur at higher frequencies than others. I have a number of DDR2 RDIMMs, mostly PC2-3200R and PC2-4200R, but no other DDR RDIMM.
This particular computer, which is supposed to support up to PC-3200R, ought to be a great platform for testing EDAC support and ECC corrected error reporting... when support for this ancient K8 EDAC controller is added, that is.

In order to gather more data, I plan on varying the memory scrubbing rate, and I suppose I could enable memory interleaving for reverting to a single NUMA node, but beyond that, I'm quite dry on ideas.
If a processor exception such as #MC had been raised, I'd have expected to get a proper reporting screen.
In general, PSU-related trouble is a possible cause for hangs and memory errors, but is it likely to be a cause when there's good timing reproducibility at varying power conditions (in configuration menu, 2C/2T in NUMA-aware mode, 4C/4T in NUMA-unaware mode) ? The computer does not crash when idling in Linux for several days.
As it stands, the time at which the black screen + hang occurs is close to 3 x the interval between ECC corrected errors, but are 3 unacknowledged EDAC reports really supposed to hang a computer ??

**Comment by x86fr** [comment]
Even back in the very first MEmtest86+ v1, Rioworks chipsets were a nightmare to support. :-/

If the black screen occurs exactly after the same amount of time, you might have triggered a watchdog.

**Comment by debrouxl** [comment]
Indeed, I thought about a watchdog, all the more since my previous message, I checked that the computer can run several days of Prime95 (2 jobs x 2 threads) without crashing: that rules out PSU-related issues. However, I didn't see a watchdog-related option in the BIOS menu, or a watchdog-related relevant entry the Linux kernel log. Next time I power the computer on, I'll double-check the kernel log and loaded modules.
I'd also need to see whether using an USB keyboard makes a difference over the PS/2 keyboard, or whether passing the `edac_report=off` parameter onto the Linux kernel cmdline triggers different behaviour.

Since my previous message, I varied memory scrubbing rates, to no avail: memtest86+ always crashes after pretty much the same amount of time, and the ECC corrected errors are still being produced at the same interval. I haven't tried to see whether memtest86+ shows errors after disabling ECC in the BOS.
The code on the numa branch is buggy when there's a single NUMA node, so reverting to a single NUMA node produces a huge number of spurious memory errors during tests, and I didn't let it run up to ~15'. I did not spot the issue quickly in the fallback (fake NUMA emulation) part of the code I added.

The only occurrence of Rioworks I can see in the Linux kernel sources is in `drivers/input/serio/i8042-x86ia64io.h`:
```
...
static const struct dmi_system_id __initconst i8042_dmi_noloop_table[] = {
        {
                /*
                 * Arima-Rioworks HDAMB -
                 * AUX LOOP command does not raise AUX IRQ
                 */
                .matches = {
                        DMI_MATCH(DMI_BOARD_VENDOR, "RIOWORKS"),
                        DMI_MATCH(DMI_BOARD_NAME, "HDAMB"),
                        DMI_MATCH(DMI_BOARD_VERSION, "Rev E"),
                },
        },
...
static int __init i8042_platform_init(void)
{
...
        if (dmi_check_system(i8042_dmi_noloop_table))
                i8042_noloop = true;
...
```

In `drivers/input/serio/i8042.c`:
```
...
static bool i8042_noloop;
module_param_named(noloop, i8042_noloop, bool, 0);
MODULE_PARM_DESC(noloop, "Disable the AUX Loopback command while probing for the AUX port");
...
(2 other occurrences)
...
```
but that's a keyboard & mouse controller. There's no occurrence of "watch" in that file, at least.

**Comment by debrouxl** [comment]
* in the kernel log, the only occurrence of "watchdog" is `NMI watchdog: Enabled. Permanently consumes one hw-PMU counter`. This watchdog can be disabled by passing `nowatchdog` to the kernel, for instance;
* using an USB keyboard instead of a PS/2 keyboard does not prevent the black screen after 15+';
* Debian 12 linux-image-6.1.0-10-amd64 doesn't hang when passing `edac_report=off nomce` on the commandline, disabling rasdaemon, and preventing the `edac_mce_amd` and `amd64_edac` modules from being loaded.
